src/sc/sidechainrpc.cpp

```diff
@@ -219,6 +227,298 @@ void AddScInfoToJSON(UniValue& result)
+    std::sort(vInputUtxo.begin(), vInputUtxo.end(), [](CmdUTXO i, CmdUTXO j) -> bool {
+        return ( std::get<2>(i) < std::get<2>(j));
+    });
+
+    CAmount totUTXOAmount = 0;
+    for (const auto & t : vInputUtxo) {
+        totUTXOAmount += std::get<2>(t);
+    }
+
+    CAmount targetAmount = _nAmount + _fee;
+
+    if (totUTXOAmount < targetAmount)
+    {
+        std::string addrDetails;
+        if (_hasFromAddress)
+            addrDetails = strprintf(" for taddr[%s]", _fromMcAddress.ToString());
+
+        throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS,
+            strprintf("Insufficient transparent funds %s, have %s, need %s (minconf=%d)",
+            addrDetails, FormatMoney(totUTXOAmount), FormatMoney(targetAmount), _minConf));
+    }
+
+    CAmount dustChange = -1;
+
+    std::vector<CmdUTXO> vSelectedInputUTXO;
+
+    for (const CmdUTXO & t : vInputUtxo)
+    {
+        _totalInputAmount += std::get<2>(t);
+        vSelectedInputUTXO.push_back(t);
+
+        if (_totalInputAmount >= targetAmount)
+        {
+            // Select another utxo if there is change less than the dust threshold.
+            dustChange = _totalInputAmount - targetAmount;
+            if (dustChange == 0 || dustChange >= _dustThreshold) {
+                break;
+            }
+        }
+    }
+
+    // If there is transparent change, is it valid or is it dust?
+    if (dustChange < _dustThreshold && dustChange != 0) {
+        throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS,
+            strprintf("Insufficient transparent funds, have %s, need %s more to avoid creating invalid change output %s (dust threshold is %s)",
+            FormatMoney(totUTXOAmount), FormatMoney(_dustThreshold - dustChange), FormatMoney(dustChange), FormatMoney(_dustThreshold)));
+    }
+
+    // Check mempooltxinputlimit to avoid creating a transaction which the local mempool rejects
+    size_t limit = (size_t)GetArg("-mempooltxinputlimit", 0);
+    if (limit > 0)
+    {
+        size_t n = vSelectedInputUTXO.size();
+        if (n > limit) {
+            throw JSONRPCError(RPC_WALLET_ERROR, strprintf("Too many transparent inputs %zu > limit %zu", n, limit));
+        }
+    }
+
+    // update the transaction with these inputs
+    for (const auto& t : vSelectedInputUTXO)
+    {
+        uint256 txid = std::get<0>(t);
+        int vout = std::get<1>(t);
+
+        CTxIn in(COutPoint(txid, vout));
+        _tx.vin.push_back(in);
+    }
+}
+
+void ScRpcCreationCmd::addChange()
+{
+    CAmount change = _totalInputAmount - ( _nAmount + _fee);
+
+    if (change > 0)
+    {
+        CReserveKey keyChange(pwalletMain);
+        CPubKey vchPubKey;
+
+        if (!keyChange.GetReservedKey(vchPubKey))
+        {
+            throw JSONRPCError(RPC_WALLET_KEYPOOL_RAN_OUT, "Could not generate a taddr to use as a change address"); // should never fail, as we just unlocked
+        }
+
+        // TODO: maybe we should send change to sender address if any?
```

Comment:
I understand this is still under discussion and it tangles with Sphere functioning. I'd vote for sending the change back to sender address

Reply:
Yes, this is an open point but I agree, a better default approach is to use the sender address for the change.